webui/src/lib/onboarding/repoOnboardingService.tsx

```diff
@@ -0,0 +1,136 @@
+/* eslint-disable react/jsx-no-target-blank */
+import React from "react";
+import { OnBoardingSteps } from "./types";
+import { NavigateFunction } from "react-router-dom";
+import { branches, retention, branchProtectionRules, objects } from "../api";
+
+const IMPORT_BRANCH_REGEX = /^_(.*)_imported$/;
+const HOOKS_PATH = "_lakefs_actions/";
+const SAMPLE_REPO_FILE_NAME = "lakes.parquet";
+
+// generate a random file name to avoid collisions for the upload preflight check
+const generateRandomFileName = (): string => {
+  const randomString = Math.random().toString(36).substring(2, 15);
+  return `sample_${randomString}.yaml`;
+};
+
+export const isSampleRepo = async (
+  repo: string,
+  defaultBranch: string,
+): Promise<boolean> => {
+  const objectList = await objects.list(repo, defaultBranch, "");
+  return objectList.results.some(
+    // when the API client is typed, we can remove the eslint-disable
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    (object: any) => object?.path === SAMPLE_REPO_FILE_NAME
+  );
+};
+
+export const canUseRepoOnboarding = async (
+  repoId: string,
+  defaultBranch: string,
+): Promise<boolean> => {
+  const promises = [
+    retention.setGCPolicyPreflight(repoId),
+    branchProtectionRules.createRulePreflight(repoId),
+    objects.list(
+      repoId,
+      defaultBranch,
+      `${HOOKS_PATH}${generateRandomFileName()}`
+    ),
+  ];
+  
+  try {
+    await Promise.all(promises);
+    return true;
+  } catch (e) {
+    return false;
+  }
+};
+
+export const getRepoOnboardingSteps = (
+  currentRepo: string,
+  defaultBranch: string,
+  objectStoreName: string,
+  navigate: NavigateFunction
+): OnBoardingSteps => [
+  {
+    id: "import-data",
+    title: "Import your existing data",
+    description: (
+      <>
+        <p>Import your existing data from ${objectStoreName} to your repository by creating pointers to it (no copying of data involved).</p>
+        <p>Please see our documentation for information on the <a href="https://docs.lakefs.io/howto/import.html#limitations" target="_blank">limitations of the import process</a> and <a href="https://docs.lakefs.io/howto/import.html#working-with-imported-data" target="_blank">working with imported data</a></p>,
+      </>
+    ),
+    cta: "Run import",
+    onClick: () =>
+      navigate(`/repositories/${currentRepo}/objects?importDialog=true`),
+    showStep: () => true, // objectStoreName !== "local",
+    isCompleted: async () => {
+      const repoBranches = await branches.list(currentRepo);
+      // when the API client is typed, we can remove the eslint-disable
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      return repoBranches.results.some((branch: any) =>
+        IMPORT_BRANCH_REGEX.test(branch.id)
+      );
```

Comment:
If a user has many branches starting with a digit or an uppercase letter, `branches.list` won't return them all -- and they'll be nagged to import data.

At the very least, list branches only with prefix `_` and open a bug to paginate this listing.

Reply:
Fixed.

Reply:
I'm confused here by "fixed".

The new code does not yet paginate.  Instead it lists a few branches with prefix `_`.  Please open a bug to paginate this listing.  This is also a good place to add a comment
```js
// BUG(#9999): Paginate!
```
or similar so that we see it when looking at the code.  Then we can know that my comment was "fixed".

Reply:
No need. Fixed with pagination in mind.